arrangor/src/main/kotlin/no/nav/amt/tiltak/arrangor/ArrangorServiceImpl.kt

```diff
@@ -46,8 +49,30 @@ class ArrangorServiceImpl(
 		return arrangorRepository.getByOrganisasjonsnummer(virksomhetsnummer)?.toArrangor()
 	}
 
-	override fun getVirksomheterForAnsatt(ansattId: UUID): List<Arrangor> {
-		throw NotImplementedError("getVirksomheterForAnsatt in ArrangorService is not yet implemented")
+	override fun oppdaterArrangor(arrangorUpdate: ArrangorUpdate) {
+		val arrangor = getArrangorByVirksomhetsnummer(arrangorUpdate.organisasjonsnummer)
+
+		if (arrangor == null) {
+			// Sjekk om virksomheten er en overordnet enhet til noen arrangører og oppdater navn.
+			arrangorRepository.updateOverordnetEnhetNavn(arrangorUpdate.organisasjonsnummer, arrangorUpdate.navn)
+			return
+		}
+
+		val overordnetEnhet = arrangorUpdate.overordnetEnhetOrganisasjonsnummer?.let {
+			enhetsregisterClient.hentVirksomhet(it)
```

Comment:
Er det ikke topicen som tilbyr informasjon om hva som evt er det nye navnet på den overordnede enheten? Rett over så antar man jo at det er en overordnet enhet og isåfall setter man overordnet enhet på alle barna. Det gjør at jeg antar at en endring i navn på overordnet enhet skal føre til en ny record på topicen

Reply:
Problemet er om det er et en ny overordnet enhet med nytt orgnr, da har vi ikke navnet og vi har ikke plukket opp endringen på topicen.